src/commands/module/add.ts

```diff
@@ -29,8 +30,12 @@ export default defineCommand({
   async setup(ctx) {
     const cwd = resolve(ctx.args.cwd || '.')
 
-    // TODO: Resolve and validate npm package name first
     const npmPackage = ctx.args.moduleName
+    const isValid = await validateNpmModule(npmPackage) 
+
+    if (!isValid) {
+      return consola.error(`\`${npmPackage}\` is not a valid npm package`)
```

Comment:
We might use `throw new Error` instead for better control flow and proper exit status code.

Reply:
aw yes, thanks